devito/grid.py

```diff
@@ -187,9 +187,9 @@ def spacing_map(self):
         return dict(zip(self.spacing_symbols, self.spacing))
 
     @property
-    def origin_domain(self):
+    def origin_offset(self):
         """
-        Origin of the local (per-process) physical domain.
+        Offset of the local (per-process) origin from the domain origin.
         """
         grid_origin = [min(i) for i in self.distributor.glb_numb]
         assert len(grid_origin) == len(self.spacing)
```

Comment:
`tuple(i*h for i, h in zip(grid_origin, self.spacing))` only correct for 0 origin

replace by:

`tuple(o.data + i*h for o, i, h in zip(self.origin, grid_origin, self.spacing))`

Reply:
no, this is supposed to be the **offset** from whatever the domain origin is
look where it's used: https://github.com/opesci/devito/blob/mpi-acoustic/devito/function.py#L1869
it's there only to translate coordinates

indeed, I tried running test_mpi::test_adjoint_F with the change you propose here, and it fails

Reply:
um ok.